tests/system.rs

```diff
@@ -63,6 +63,30 @@ fn coverage_tests() {
     assert!(fails == 0, "{} tests failed", fails);
 }
 
+#[test]
+fn checkstyle_test() {
+    let filename = "tests/source/fn-single-line.rs".to_string();
+    let expected = "tests/writemode/checkstyle.xml";
```

Comment:
My thinking was that we could have one test for each write-mode that takes compares the output of `rustfmt` against an input file. I could extract the expected/actual output into a separate function that will be easier to reuse.